### PR TITLE
vdk-core: add vdk sdk secrets api - part I

### PR DESCRIPTION
**Pull request by dakodakov**
Incremental change for adding the secrets capability.

Since we have a somewhat circular dependecy with the vdk-test-utils plugin more changes - full secerts implementation and tests will follow in the next MR.

### CODE REVIEW

File: projects/vdk-core/src/vdk/api/plugin/plugin_input.py [hunk 1]
```diff
@@ -77,13 +77,72 @@ def set_properties_factory_method(
...
+        :param secrets: dictionary with the new job's secrets
+        :return secrets: dictionary with the secrets handled
+        """
+        pass
+
+
+ISecretsFactory = Callable[[], ISecretsServiceClient]
+
+
+class ISecretsRegistry(ABC):
```
Comment (murphp15) [inline projects/vdk-core/src/vdk/api/plugin/plugin_input.py:119]: will there be multiple implementations of this?
I wouldn't have thought this needed to be an interface.

**Review by murphp15 — APPROVED** [review]

File: projects/vdk-core/src/vdk/api/plugin/plugin_input.py (hunk 1 above)
Comment (dakodakov) [reply projects/vdk-core/src/vdk/api/plugin/plugin_input.py:119]: Yes, it's pretty much copied from the Properties API where we have "the real" one as well as test/dummy/in-memory implementations... Additionally it allows for extensibility.

**Review by mivanov1988 — APPROVED** [review]

**Review by tozka — APPROVED** [review]

File: projects/vdk-core/src/vdk/api/job_input.py [hunk 2]
```diff
@@ -33,6 +33,26 @@ def set_all_properties(self, properties: dict):
...
 
+class ISecrets:
+    """
+    Allows for Data Job to store and retrieve sensitive data.
+    Secrets are solution for the following use cases
+    * Keeping API keys and passwords necessary to connect to different systems
+    """
+
+    @abstractmethod
+    def get_secret(self, name: str, default_value: Any = None) -> str:
```
Comment (tozka) [inline projects/vdk-core/src/vdk/api/job_input.py:44]: nit: name of what? I know it's also not documented in get_property but I was left wondering what is the expected format of name , just any string. 

Btw. what are the allow value formats. They are also not documented explicitly in IProperties but there are checks. Perhaps they should have been

It's a nit (feel free to ignore comment).

File: projects/vdk-core/src/vdk/api/plugin/plugin_input.py [hunk 3]
```diff
@@ -77,13 +77,72 @@ def set_properties_factory_method(
...
+    """
+    Secrets registry to enable register implementation Secrets service
```
Comment (tozka) [inline projects/vdk-core/src/vdk/api/plugin/plugin_input.py:121]: nit: that sounds awkward to me 

```suggestion
        Registry to enable the registration of Secrets service implementations.
```

File: projects/vdk-core/src/vdk/api/job_input.py (hunk 2 above)
Comment (dakodakov) [reply projects/vdk-core/src/vdk/api/job_input.py:44]: will address them it in the upcoming MR

**Pull request merged**